src/jenkins.py

```diff
@@ -269,9 +270,22 @@ def _configure_proxy(
 
     client = client if client is not None else _get_client(get_admin_credentials(container))
 
-    proxy_args = [f"'{proxy_config.hostname}'", f"'{proxy_config.port}'"]
-    if proxy_config.username and proxy_config.password:
-        proxy_args += [f"'{proxy_config.username}'", f"'{proxy_config.password}'"]
+    if proxy_config.https_proxy:
+        proxy_args = [
+            f"'{proxy_config.https_proxy.host}'",
+            f"{proxy_config.https_proxy.port}",
+            f"'{proxy_config.https_proxy.user or ''}'",
+            f"'{proxy_config.https_proxy.password or ''}'",
+        ]
+    else:
+        # http proxy and https proxy value both cannot be None at the same time.
```

Comment:
This is confusing? What does this mean?

Reply:
Would 
```suggestion
        # http proxy and https proxy value cannot both be None since proxy_config would be parsed as None.
```
be better?

Reply:
ah yes

Reply:
Fixed, thank you!